src/components/Message/MessageCommerce.js

```diff
@@ -355,81 +107,200 @@ class MessageCommerce extends PureComponent {
-                    reaction_counts={message.reaction_counts}
-                    latest_reactions={message.latest_reactions}
-                    ref={this.reactionSelectorRef}
-                  />
-                )}
-              </React.Fragment>
-            )}
-
-            {message?.attachments &&
-              (!images || images.length <= 1) &&
-              message.attachments.map(
-                /** @type {(item: import('stream-chat').Attachment) => React.ReactElement | null} Typescript syntax */
-                (attachment, index) => (
-                  // @ts-ignore Attachment is not yet typed
-                  <Attachment
-                    key={`${message.id}-${index}`}
-                    attachment={attachment}
-                    actionHandler={handleAction}
-                  />
-                ),
+      >
+        {(firstGroupStyle === 'bottom' || firstGroupStyle === 'single') && (
+          <Avatar
+            image={message?.user?.image}
+            size={32}
+            name={message?.user?.name || message?.user?.id}
+            onClick={onUserClick}
+            onMouseOver={onUserHover}
+          />
+        )}
+        <div className="str-chat__message-commerce-inner">
+          {message && !message.text && (
+            <React.Fragment>
+              {
+                <MessageOptions
+                  {...props}
+                  displayLeft={false}
+                  displayReplies={false}
+                  displayActions={false}
+                  onReactionListClick={onReactionListClick}
+                  theme={'commerce'}
+                />
+              }
+              {/* if reactions show them */}
+              {hasReactions && !showDetailedReactions && (
+                <ReactionsList
+                  reactions={message.latest_reactions}
+                  reaction_counts={message.reaction_counts}
+                  onClick={onReactionListClick}
+                />
+              )}
+              {showDetailedReactions && (
+                <ReactionSelector
+                  reverse={false}
+                  handleReaction={handleReaction || propHandleReaction}
+                  actionsEnabled={actionsEnabled}
+                  detailedView
+                  reaction_counts={message.reaction_counts}
+                  latest_reactions={message.latest_reactions}
+                  ref={reactionSelectorRef}
+                />
               )}
-            {images && images.length > 1 && <Gallery images={images} />}
+            </React.Fragment>
+          )}
 
-            {message?.text && this.renderMessageText()}
-            {!threadList && (
-              <div className="str-chat__message-commerce-reply-button">
-                <MessageRepliesCountButton
-                  onClick={handleOpenThread}
-                  reply_count={message?.reply_count}
+          {message?.attachments &&
+            (!images || images.length <= 1) &&
```

Comment:
probably useless check

Reply:
It does have an effect. For example, if the message has 2 pdfs and 2 jpgs, we want (as in, that was the behavior) to display just the jpgs on a Gallery, ignoring the other attachments.

Reply:
i know it's the old logic but I still don't see how the image check relates here, especially that we check for falsy, zero or 1 length value

Reply:
Should we remove it, and in the example I gave where you have two pdfs and two jpgs, instead of just rendering one gallery with the images, we would render 4 `<Attachment />` components **and** the gallery in the same message.